pkg/kubelet/cm/container_manager_linux_test.go

```diff
@@ -268,3 +268,65 @@ func TestGetCapacity(t *testing.T) {
 		})
 	}
 }
+
+func TestNewPodContainerManager(t *testing.T) {
+	cases := []struct {
+		name string
+		cm   *containerManagerImpl
+	}{
+		{
+			name: "CgroupsPerQOS is disabled, return *podContainerManagerNoop",
+			cm: &containerManagerImpl{
+				qosContainerManager: &qosContainerManagerImpl{
+					qosContainersInfo: QOSContainersInfo{
+						Guaranteed: CgroupName{"guaranteed"},
+						BestEffort: CgroupName{"besteffort"},
+						Burstable:  CgroupName{"burstable"},
+					},
+					subsystems:         &CgroupSubsystems{},
+					cgroupManager:      NewCgroupManager(&CgroupSubsystems{}, ""),
+					activePods:         func() []*v1.Pod { return []*v1.Pod{} },
+					getNodeAllocatable: func() v1.ResourceList { return v1.ResourceList{} },
+					cgroupRoot:         CgroupName{"burstable"},
+					qosReserved:        map[v1.ResourceName]int64{},
```

Comment:
Do we really need to fill all these fields to test object creation?

Reply:
good point. Fixed and added another two tests to also cover the `useSystemd` scenario